src/ImageSharp/Processing/Processors/Convolution/KernelSamplingMap.cs

```diff
@@ -105,5 +81,79 @@ public void Dispose()
                 this.isDisposed = true;
             }
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void BuildOffsets(IMemoryOwner<int> offsets, int boundsSize, int kernelSize, int min, int max, BorderWrappingMode borderMode)
+        {
+            int radius = kernelSize >> 1;
+            Span<int> span = offsets.GetSpan();
+            ref int spanBase = ref MemoryMarshal.GetReference(span);
+            for (int chunk = 0; chunk < boundsSize; chunk++)
+            {
+                int chunkBase = chunk * kernelSize;
+                for (int i = 0; i < kernelSize; i++)
+                {
+                    Unsafe.Add(ref spanBase, chunkBase + i) = chunk + i + min - radius;
+                }
+            }
+
+            this.CorrectBorder(span, kernelSize, min, max, borderMode);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void CorrectBorder(Span<int> span, int kernelSize, int min, int max, BorderWrappingMode borderMode)
+        {
+            var affectedSize = (kernelSize >> 1) * kernelSize;
+            if (affectedSize > 0)
+            {
+                switch (borderMode)
+                {
+                    case BorderWrappingMode.Repeat:
+                        Numerics.Clamp(span.Slice(0, affectedSize), min, max);
+                        Numerics.Clamp(span.Slice(span.Length - affectedSize), min, max);
+                        break;
+                    case BorderWrappingMode.Mirror:
+                        for (int i = 0; i < affectedSize; i++)
+                        {
+                            var value = span[i];
+                            if (value < min)
+                            {
+                                span[i] = min - value + min;
+                            }
+                        }
+
+                        for (int i = span.Length - affectedSize; i < span.Length; i++)
+                        {
+                            var value = span[i];
+                            if (value > max)
+                            {
+                                span[i] = max - value + max;
+                            }
+                        }
+
+                        break;
+                    case BorderWrappingMode.Wrap:
+                        for (int i = 0; i < affectedSize; i++)
+                        {
+                            var value = span[i];
+                            if (value < min)
+                            {
+                                span[i] = max - min + value + 1;
+                            }
+                        }
+
+                        for (int i = span.Length - affectedSize; i < span.Length; i++)
+                        {
+                            var value = span[i];
+                            if (value > max)
+                            {
+                                span[i] = min + value - max - 1;
```

Comment:
Same caching of `min - max - 1` as above.

Reply:
Done, with same variable as above.